### PR TITLE
CIT-41 #time 5h Implment Like functionality in the blog app, user can…

### PR DESCRIPTION
**Pull request by codal-uchoudhary**
… use single API to like or dislike the blog

### CODE REVIEW

**Comment by codal-uchoudhary** [comment]
- Create API to like or dislike the blog (Using SINGLE API)
- Create API to get all the users who like the specific blog.

File: Projects/My_website/myWebsite/account/views.py [hunk 1]
```diff
@@ -32,20 +32,23 @@ def post(self, request):
...
+                )
             token = Token.objects.get_or_create(user=user)
+            print(type(token))
             return Response(
                 {
                     "status": True,
                     "message": "user is loged",
-                    "token": str(token),
+                    "token": str(token[0]),
                     "first_name": user.first_name,
```
Comment (arvish-codal) [inline Projects/My_website/myWebsite/account/views.py]: @codal-uchoudhary 
1) Remove unnecessary print statements
2) don't write token[0]. Find a better approach.

File: Projects/My_website/myWebsite/blog/views.py [hunk 2]
```diff
@@ -105,20 +104,33 @@ def partial_update(self, request, pk):
...
+            blog = Post.objects.get(id=request.data["blog-id"])
+            obj = blog.like.filter(id=request.user.id)
+            if obj.exists():
+                blog.like.remove(request.user.id)
+                return Response({"you disliked the blog"})
+            blog.like.add(request.user.id)
+            blog.save()
+            return Response({"you liked this blog"})
+        except:
+            return Response({"message": "Provided blog-id is not valid"})
```
Comment (arvish-codal) [inline Projects/My_website/myWebsite/blog/views.py]: @codal-uchoudhary 
Don't provide a static response. Raise the exception.

File: Projects/My_website/myWebsite/blog/views.py [hunk 3]
```diff
@@ -105,20 +104,33 @@ def partial_update(self, request, pk):
...
-        serializer = UserSerializer(data, many=True)
-        return Response(serializer.data)
+    def get(self, request):
+        try:
+            blog = Post.objects.get(id=request.data["blog-id"])
+            people = blog.like.all()
+            serializer = UserSerializer(people, many=True)
+            return Response(serializer.data)
+        except:
+            return Response({"message": "provided blog-ID in not valid"})
```
Comment (arvish-codal) [inline Projects/My_website/myWebsite/blog/views.py]: @codal-uchoudhary 
same here. Raise the exception

File: Projects/My_website/myWebsite/account/views.py (hunk 1 above)
Comment (codal-uchoudhary) [reply Projects/My_website/myWebsite/account/views.py]: Implemented

File: Projects/My_website/myWebsite/blog/views.py (hunk 2 above)
Comment (codal-uchoudhary) [reply Projects/My_website/myWebsite/blog/views.py]: Implemented

File: Projects/My_website/myWebsite/blog/views.py (hunk 3 above)
Comment (codal-uchoudhary) [reply Projects/My_website/myWebsite/blog/views.py]: implemented

File: Projects/My_website/myWebsite/blog/views.py [hunk 4]
```diff
@@ -119,7 +119,7 @@ def post(self, request):
             blog.save()
             return Response({"you liked this blog"})
         except:
-            return Response({"message": "Provided blog-id is not valid"})
+            raise Exception("provided blog-id is not valid")
```
Comment (arvish-codal) [inline Projects/My_website/myWebsite/blog/views.py]: @codal-uchoudhary 
It should be something like this

except Exception as e:
  raise e

File: Projects/My_website/myWebsite/blog/views.py [hunk 5]
```diff
@@ -133,4 +133,4 @@ def get(self, request):
             serializer = UserSerializer(people, many=True)
             return Response(serializer.data)
         except:
-            return Response({"message": "provided blog-ID in not valid"})
+            raise Exception("provided blog-id is not valid")
```
Comment (arvish-codal) [inline Projects/My_website/myWebsite/blog/views.py]: @codal-uchoudhary 

same to be done here as well.

**Comment by arvish-codal** [comment]
@codal-uchoudhary ,
am approving and merging for now.

but pls make sure you don't forget to add those changes.

**Comment by codal-uchoudhary** [comment]
Code is Updated as per comments.

**Review by arvish-codal — APPROVED** [review]

**Pull request merged**